CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -363,8 +363,12 @@ public static function postProcess(&$form) {
     $skipDeceased = isset($form->skipDeceased) ? $form->skipDeceased : TRUE;
     $html = $activityIds = array();
 
+    // CRM-21255 - Hrm, CiviCase 4+5 seem to report buttons differently...
+    $c = $form->controller->container();
+    $isLiveMode = ($buttonName == '_qf_PDF_upload') || isset($c['values']['PDF']['buttons']['_qf_PDF_upload']);
+
     // CRM-16725 Skip creation of activities if user is previewing their PDF letter(s)
-    if ($buttonName == '_qf_PDF_upload') {
+    if ($isLiveMode) {
```

Comment:
Is the "new popup behaviour" in civicase meant to be different from the existing crm-popup stuff?  Or just an anomaly.  The answer to the first question should answer whether this fix is appropriate or not.
Also, when I read the code, it's not clear what $isLiveMode actually means. (I know it means we are submitting rather than previewing but the variable name doesn't give me that clarity).  A variable named $isPreview would be much clearer to me, though you'd need to reverse the logic :-)

Reply:
To me, it feels like a bug in the new popup code, but @colemanw is really the one who's worked with that bit, so I wanted to get his read on whether/how to fix that.

If we are going to proceed with this patch, then I'm happy to reverse the name/logic. (My gut reaction was originally to name it `$isPreview`, but then felt it less risky to keep the conditionals. If your gut also says `$isPreview`, then let's listen to our grumbling tummies.)

Reply:
Either that, or rename `$isLiveMode` to `$isNotPreview` to avoid reversing the existing conditionals.  A little bit wierd, but at least the next dev will find it easier to understand!